lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
Just moving explaining it in code. 

```
Router.map(function () {
  this.route('lazy-loaded-route', {path: '/lazy-loaded-route/:post_id'});
})

// path/routes/lazy-loaded-route
export default Ember.Route.extend({
  serialize: function () {
   return { post_id: 42 };
  }
});

{{link-to 'lazy-loaded-route' model}}
```

If the route is loaded, the link will be `lazy-loaded-route/42`, if the route is not loaded, the link will be `lazy-loaded-route/{model.id}`